tests/predicates_inferred/sanity-ordering.js

```diff
@@ -10,6 +10,9 @@ function dotAccess(head: { page: ?Object; }, create?: mixed) {
   const stack = path.split('.');
   do {
     const key = stack.shift();
+    if (!key) {
+      break;
+    }
```

Comment:
I'll admit that this code is worse to write and in practice the falsey check isn't necessary for _this_ code (since `obj[undefined] === undefined`). But it does show a practical way to fix the issue that's surfaced.